### PR TITLE
User FlightDataEncoder from arrow-rs

### PR DESCRIPTION
**Pull request by thinkharderdev**
[VTX-1203]

Switch the executor flight server to use `FlightDataEncoder` from `arrow-rs` which already handles splitting batches to ensure a max gRPC message size. 

[VTX-1203]: https://coralogix.atlassian.net/browse/VTX-1203?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: ballista/scheduler/src/state/execution_graph.rs [hunk 1]
```diff
@@ -504,7 +504,7 @@ impl ExecutionGraph {
                                 if let Err(err) =
                                     running_stage.update_task_metrics(operator_metrics)
                                 {
-                                    error!(job_id = self.job_id, stage_id = running_stage.stage_id, error = %err, "Error updating task metrics");
+                                    warn!(job_id = self.job_id, stage_id = running_stage.stage_id, error = %err, "Error updating task metrics");
```
Comment (thinkharderdev) [inline ballista/scheduler/src/state/execution_graph.rs:507]: This should remove a lot of the zero error alerts noise

File: ballista/executor/src/cpu_bound_executor.rs [hunk 2]
```diff
@@ -146,7 +146,7 @@ impl DedicatedExecutor {
         let job = Box::pin(async move {
             let task_output = task.await;
             if tx.send(task_output).is_err() {
-                warn!("Spawned task output ignored: receiver dropped");
+                debug!("Spawned task output ignored: receiver dropped");
```
Comment (thinkharderdev) [inline ballista/executor/src/cpu_bound_executor.rs:149]: Not really causing any noise but annoying when searching for `WARN`ings and there are a million of these to wade through...

Comment (iravid) [reply ballista/executor/src/cpu_bound_executor.rs:149]: 💯

File: ballista/executor/src/executor_server.rs [hunk 3]
```diff
@@ -65,6 +65,10 @@ use crate::scheduler_client_registry::SchedulerClientRegistry;
 use crate::shutdown::ShutdownNotifier;
 use crate::{as_task_status, TaskExecutionTimes};
 
+// Set the max gRPC message size to 64 MiB. This is quite large
+// but we have to send execution plans over gRPC and they can be large.
+const MAX_GRPC_MESSAGE_SIZE: usize = 64 * 1024 * 1024;
```
Comment (iravid) [inline ballista/executor/src/executor_server.rs:70]: What's the difference between this and the 4MB constant in the other file?

Comment (iravid) [reply ballista/executor/src/executor_server.rs:70]: Cool thanks

**Review by iravid — APPROVED** [review]

**Review by Dandandan — APPROVED** [review]
LGTM

**Pull request merged**